source/scpd/Cpp.d

```diff
@@ -192,40 +192,39 @@ extern(C++, (StdNamespace)) extern(C++, class) struct vector (T, Alloc = allocat
         /// TODO: Separate from `vector` definition
         private static struct ConstIterator
         {
-            const(T)* ptr;
+            size_t index;
             const(vector!T)* orig;
 
-            public ref const(T) front () const pure nothrow @nogc
+            public ref const(T) front () const pure nothrow @nogc @trusted
             {
-                if (this.empty)
-                    assert(0);
-                return *this.ptr;
+                return (*this.orig)[this.index];
             }
-            public void popFront () pure nothrow @nogc
+            public void popFront () pure nothrow @nogc @trusted
             {
                 if (!this.empty)
-                    this.ptr++;
+                    this.index++;
             }
             public @property bool empty () const pure nothrow @safe @nogc
             {
-                return !(this.ptr < this.orig._end);
+                return !(this.index < this.orig.length);
             }
         }
 
-        public ref inout(T) opIndex(size_t idx) inout pure nothrow @nogc
+        public ref inout(T) opIndex(size_t idx) inout pure nothrow @nogc @trusted
         {
-            assert(idx < this.length);
+            if (idx >= this.length)
```

Comment:
Do we want to start using `verify` from ocean?

Reply:
I don't, do you ?

Reply:
I was just thinking it simplifies the expression. Because instead of testing "if something is out of range, throw" it tests "this must be in this specific range".